traitsui/tests/editors/test_font_editor.py

```diff
@@ -0,0 +1,53 @@
+#  Copyright (c) 2005-2020, Enthought, Inc.
+#  All rights reserved.
+#
+#  This software is provided without warranty under the terms of the BSD
+#  license included in LICENSE.txt and may be redistributed only
+#  under the conditions described in the aforementioned license.  The license
+#  is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+#  Thanks for using Enthought open source!
+#
+""" Test FontEditor
+"""
+import unittest
+
+from traits.api import HasTraits
+from traitsui.api import Font, Item, View
+from traitsui.tests._tools import (
+    is_wx,
+    requires_toolkit,
+    ToolkitName,
+)
+from traitsui.testing.tester import command
+from traitsui.testing.tester.ui_tester import UITester
+from traitsui.testing.tester.registry import TargetRegistry
+
+
+class ObjectWithFont(HasTraits):
+    font_trait = Font()
+
+
+# A local registry of interaction handlers for testing purposes.
+LOCAL_TARGET_REGISTRY = TargetRegistry()
+
+if is_wx():
+    from traitsui.wx.font_editor import TextEditor
+    LOCAL_TARGET_REGISTRY.register_handler(
+        target_class=TextEditor,
+        interaction_class=command.MouseClick,
+        handler=lambda wrapper, _: wrapper.target.control.SetFocus()
+    )
```

Comment:
This demonstrates how we can write tests for corner cases without necessarily exposing those test logic to the tester public API.
However this particular interaction might be worth bumping to the public API anyway.
After #1190, I will be able to extend support for the text-style font editor more trivially.